### ISSUE TITLE
systemd unit for tango

### ISSUE THREAD

**Issue by picca** [tango-controls/TangoSourceDistribution#58]
Hello, it would be nice to have systemd files for tango-db, tango-starter, and tango-accesscontrol.

**Comment by t-b** [comment]
@sergirubio Has some systemd unit files from the debian packages they are building.

**Comment by sergirubio** [comment]
Hi,

I forward the answer from Carlos Falcon (cfalcon@cells.es)

     "We use and contribute to the official debian scripts (init.d) I 
show in the tango doc examples of systemd scripts created by Igors"

**Comment by bourtemb** [comment]
Hi @sergirubio and @cmft , any update on this topic?

**Comment by t-b** [comment]
Now I'm understanding the comment of @cmft.

There is an example in https://tango-controls.readthedocs.io/en/latest/tutorials-and-howtos/how-tos/how-to-integrate-with-systemd.html?highlight=systemd with same bare bones version of systemd files.

@picca Is that a good enough starting point for you? One thing missing from these is parsing `/etc/tangorc` for TANGO_HOST.

**Comment by mliszcz** [comment]
> 
> @picca Is that a good enough starting point for you? One thing missing from these is parsing `/etc/tangorc` for TANGO_HOST.

Why do you want to parse the tangorc in the first place? The client does the parsing if environment variable is not set. See: https://github.com/tango-controls/cppTango/blob/81286b2eab1ec27686f9428f3efa03681f5a3164/cppapi/client/dbapi_base.cpp#L69

**Comment by t-b** [comment]
@mliszcz Because that is what the current init.d starting scripts in debian do. The database server needs to know at what address it should listen.

**Comment by picca** [comment]
I do not know a lot about systemd, but it is possible to set a bunch of environment variable before starting a process.
A nice feature would be also to instrument the code in order to add the systemd notify. this way we will have a more reliable service.

another solution would be to use a wrapper in order to start the process.

Maybe it is now enought to se the right /etc/tangorc and let the Databse Device use this information to start on the right interface and port.

after, I do not know if we should be able to start multiple database server on the same host but listening on different interfaces.

https://www.freedesktop.org/software/systemd/man/sd_notify.html

**Comment by t-b** [comment]
https://tango-controls.readthedocs.io/en/latest/tutorials-and-howtos/how-tos/how-to-try-tango/tango-db-service.html has an example using `EnvironmentFile=`, see also https://www.freedesktop.org/software/systemd/man/systemd.exec.html#EnvironmentFile=. This directive is able to read our `/etc/tangorc`.
